network/src/peers/peer_book.rs

```diff
@@ -266,6 +266,10 @@ impl PeerBook {
         debug!("Added {} to the peer book", address);
     }
 
+    pub async fn remove_peer(&self, address: SocketAddr) -> Option<Peer> {
```

Comment:
We might want to comment on the fact that this function is only used in tests; otherwise it would make sense to first check the collection of connected peers and perform a disconnect first (which is not needed for test purposes).

Alternatively, the function could be renamed to `remove_disconnected` peer, which would make its purpose clear as well.